src/middleware.page.ts

```diff
@@ -0,0 +1,16 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+// run only on homepage
+export const config = {
+  matcher: '/',
+};
+
+export async function middleware(req: NextRequest) {
+  const { nextUrl: url, geo } = req;
+  console.log(geo);
+  if (!url.searchParams.has('lat') && geo && geo.latitude && geo.longitude) {
+    url.searchParams.append('lat', geo.latitude);
+    url.searchParams.append('lng', geo.longitude);
```

Comment:
I'm not sure I agree with appending this to the `searchParams`.

A user who is on the go and using their mobile might reload the page at some point, but because lat/lng are already set as params they won't update to the latest edge location.

Also, people who share the link to the site with their friends will likely bring the lat/lng along if they copy/paste the link from their browser meaning that people who land on the site will get the wrong initial location for them